plugins/package-managers/nuget/src/main/kotlin/utils/NuGetInspector.kt

```diff
@@ -0,0 +1,304 @@
+    }
+    result.packages.first().errors.forEach {
+        errorMessage += it + "\n"
+    }
+    result.dependencies.forEach { dep -> dep.errors.forEach { errorMessage += it + "\n" } }
+    return errorMessage
+}
+
+internal fun NuGetInspector.Result.toOrtProject(
+    managerName: String,
+    analysisRoot: File,
+    definitionFile: File,
+): Project {
+    val id = Identifier(
+        type = managerName,
+        namespace = "",
+        name = definitionFile.relativeTo(analysisRoot).invariantSeparatorsPath,
+        version = ""
+    )
+    val workingDir = definitionFile.parentFile
+
+    val nestedPackages: MutableList<NuGetInspector.PackageData> = mutableListOf()
+
+    packages.forEach {
+        it.dependencies.forEach { pkg -> nestedPackages.add(pkg) }
+    }
+
+    val packageReferences = nestedPackages.toPackageReferences()
+
+    val scopes = sortedSetOf(Scope(headers.first().projectFramework, packageReferences))
+    return Project(
+        id = id,
+        definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+        vcs = VcsInfo.EMPTY,
+        authors = setOf(),
+        vcsProcessed = PackageManager.processProjectVcs(workingDir),
+        declaredLicenses = setOf(),
+        homepageUrl = "",
+        scopeDependencies = scopes
+    )
+}
+
+private fun List<NuGetInspector.PackageData>.toPackageReferences(): SortedSet<PackageReference> =
+    mapTo(sortedSetOf()) { it.toPackageReference() }
+
+private fun NuGetInspector.PackageData.toPackageReference() =
+    PackageReference(
+        id = Identifier(type = TYPE, namespace = "", name = name, version = version ?: ""),
+        dependencies = dependencies.toPackageReferences()
+    )
+
+internal fun Set<NuGetInspector.PackageData>.toOrtPackages(): Set<Package> =
+    groupBy { "${it.name}:${it.version}" }.mapTo(mutableSetOf()) { (_, packages) ->
+        val pkg = packages.first()
+
+        fun NuGetInspector.PackageData.getHash(): Hash = Hash.create(
+            sha512?.lowercase() ?: sha256?.lowercase() ?: sha1?.lowercase() ?: md5?.lowercase() ?: ""
+        )
+
+        val id = Identifier(
+            type = TYPE,
+            namespace = pkg.namespace.orEmpty(),
+            name = pkg.name,
+            version = pkg.version ?: ""
+        )
+        val declaredLicenses: SortedSet<String> = sortedSetOf()
+        val pkgDeclaredLicense = pkg.declaredLicense.orEmpty()
+
+        if (pkgDeclaredLicense.isNotBlank()) {
+            val licenseData = yamlMapper.readValue<Map<String, String>>(pkgDeclaredLicense)
+            val type = licenseData["LicenseType"].orEmpty()
+            listOfNotNull(
+                licenseData["LicenseExpression"]?.takeIf { type.lowercase() == "expression" },
+                licenseData["LicenseUrl"]
+            ).firstOrNull()?.also { declaredLicenses += it }
+        }
+
+        var vcsUrl = pkg.vcsUrl.orEmpty()
+        var commit = ""
+
+        if (vcsUrl.contains("@")) {
```

Comment:
Write as `"@" in vcsUrl`.

Reply:
Or actually, you can avoid `contains` / `in` like

        val segments = vcsUrl.split("@", limit = 2)
        if (segments.size == 2) {
            vcsUrl = segments[0]
            commit = segments[1]
        }